src/transformers/trainer.py

```diff
@@ -537,8 +547,19 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
                 lengths = None
             model_input_name = self.tokenizer.model_input_names[0] if self.tokenizer is not None else None
             if self.args.world_size <= 1:
+                if version.parse(torch.__version__) < version.parse("1.6.0"):
```

Comment:
perhaps adding `has_torch_generator` to file_utils.py?

Reply:
then don't need comments either about when it was introduced

Reply:
Used a private variable like for native AMP (it's actually the same PyTorch version check).

Reply:
That's perfect.